### PR TITLE
106-add-logger-configuration

### PR DESCRIPTION
**Pull request by l4mby**

### CODE REVIEW

**Review by gpad — COMMENTED** [review]
I think you could at least move winston from dependencies to devDependencies...
Is it possible to completely remove it?

File: test/e2e/declare_consumer.test.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ describe("declare consumer", () => {
     await eventually(() => expect(messages).eql([Buffer.from("hello"), Buffer.from("world"), Buffer.from("world")]))
   }).timeout(10000)
 
-  it(`consume a lot of messages`, async () => {
+  it.skip(`consume a lot of messages`, async () => {
```
[the remark this answers is not in the document]
Comment (albertobarrila) [reply test/e2e/declare_consumer.test.ts:75]: yes it's fixxed in the https://github.com/coders51/rabbitmq-stream-js-client/pull/112

File: src/amqp10/messageHeader.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export class Header {
           const formatCode = dataResponse.readUInt8()
           dataResponse.rewind(1)
           const decodedBoolean = decodeFormatCode(dataResponse, formatCode)
-          if (!decodedBoolean) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
+          if (decodedBoolean === undefined) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
```
Comment (gpad) [inline src/amqp10/messageHeader.ts:15]: Why? I strongly suggest to never check directly `undefined` or `null` I think the previous code is much better

File: src/response_decoder.ts [hunk 3]
```diff
@@ -264,12 +264,12 @@ function decodeMessageHeader(dataResponse: DataReader) {
...
 
-  const nextType = dataResponse.readInt8()
-  decodeFormatCode(dataResponse, nextType)
+  // next, the composite type is encoded as an AMQP uint8
+  dataResponse.readUInt64()
 
   const formatCode = dataResponse.readUInt8()
   const headerLength = decodeFormatCode(dataResponse, formatCode)
-  if (!headerLength) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
+  if (headerLength === undefined) throw new Error(`invalid formatCode %#02x: ${formatCode}`)
```
Comment (gpad) [inline src/response_decoder.ts:272]: again

File: test/support/fake_data.ts [hunk 4]
```diff
@@ -42,5 +42,7 @@ export async function createConnection(username: string, password: string, liste
     frameMax: 0, // not used
     heartbeat: 0,
     listeners: listeners,
-  })
+  }
+
+  return await connect(params)
```
Comment (gpad) [inline test/support/fake_data.ts:47]:
```typescript
export function createConnection(username: string, password: string, listeners?: ListenersParams) {
  return connect({
    hostname: "localhost",
    port: 5552,
    username,
    frameMax: 0, 
    heartbeat: 0,
    listeners: listeners,
  })
}

```

File: src/response_decoder.ts [hunk 5]
```diff
@@ -264,12 +264,12 @@ function decodeMessageHeader(dataResponse: DataReader) {
     throw new Error(`invalid composite header %#02x: ${type}`)
   }
 
-  const nextType = dataResponse.readInt8()
-  decodeFormatCode(dataResponse, nextType)
+  // next, the composite type is encoded as an AMQP uint8
+  dataResponse.readUInt64()
```
Comment (gpad) [inline src/response_decoder.ts:268]: Are you sure about it?

File: src/amqp10/messageHeader.ts (hunk 2 above)
Comment (l4mby) [reply src/amqp10/messageHeader.ts:15]: The problem is that in the previous code even if the decodedBoolean is false it throws an exception, but false is correct

File: src/response_decoder.ts (hunk 3 above)
Comment (l4mby) [reply src/response_decoder.ts:272]: Same as above, if headerLength is 0 it throws the exception, but 0 can be a correct value

File: src/response_decoder.ts (hunk 5 above)
Comment (l4mby) [reply src/response_decoder.ts:268]: This is also done in the dotnet client 
https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/RabbitMQ.Stream.Client/AMQP/AmqpWireFormattingRead.cs#L389

**Review by gpad — APPROVED** [review]

**Pull request merged**